src/apis/source.rs

```diff
@@ -376,7 +375,7 @@ impl SourceAPI {
         data_connection_id: &String,
     ) -> Result<DataConnection, BackendError> {
         // Try to get the cached value
-        let cache_key = format!("{}", data_connection_id);
+        let cache_key = data_connection_id.to_string();
```

Comment:
Calling `.to_string()` on a `String` feels odd. I think what we're really doing here is:

```suggestion
        let cache_key = data_connection_id.clone();
```